File: scripts/generate-session.ts
import "dotenv/config";
import { TelegramClient } from "telegram";
import { StringSession } from "telegram/sessions/index.js";
// @ts-ignore
import input from "input";

const apiId = Number(process.env.TG_API_ID);
const apiHash = process.env.TG_API_HASH!;
const stringSession = new StringSession("");

if (!apiId || !apiHash) {
  throw new Error("❌ TG_API_ID або TG_API_HASH не завантажились");
}

(async () => {
  const client = new TelegramClient(stringSession, apiId, apiHash, {
    connectionRetries: 5,
  });

  try {
    await client.start({
      phoneNumber: async () => {
        let number = await input.text("📱 Введи номер з +380: ");
        number = number.replace(/[^\d+]/g, "");
        if (!number.startsWith("+")) number = "+" + number;
        console.log(`✅ Номер: ${number}`);
        return number;
      },

      phoneCode: async () => {
        console.log("⏳ Чекаємо код з SMS/Telegram...");
        const code = await input.text("📨 Введи код: ");
        return code.trim();
      },

      password: async () => {
        console.log("🔐 Потрібен пароль 2FA");
        const password = await input.text("Введи пароль: ");
        return password;
      },

      onError: (err) => {
        console.error("❌ Помилка:", err.message || err);
      },
    });

    console.log("✅ Успішно увійшов!");
    const me = await client.getMe();
    console.log(`👤 Користувач: ${me.firstName}`);

    console.log("\n🔥 ТВІЙ SESSION STRING:\n");
    console.log(client.session.save());

    await client.sendMessage("me", { message: "Привіт!" });

  } catch (error) {
    console.error("💥 Помилка:", error);
  }
})();

File: scripts/generate-session.py
# scripts/generate-session.py
import asyncio
from telethon import TelegramClient
from telethon.sessions import StringSession
import os
from dotenv import load_dotenv

load_dotenv()

api_id = int(os.getenv("TG_API_ID"))
api_hash = os.getenv("TG_API_HASH")
phone = input("📱 Введи номер +380: ")

async def main():
    client = TelegramClient(StringSession(), api_id, api_hash)
    
    await client.connect()
    
    if not await client.is_user_authorized():
        await client.send_code_request(phone)
        code = input("📨 Введи код: ")
        
        try:
            await client.sign_in(phone, code)
        except Exception as e:
            password = input("🔐 Введи пароль: ")
            await client.sign_in(password=password)
    
    print("✅ Увійшов!")
    print(f"📋 Session string:\n{client.session.save()}")
    
    await client.disconnect()

if __name__ == "__main__":
    asyncio.run(main())